packages/dd-trace/src/telemetry/dependencies.js

```diff
@@ -0,0 +1,81 @@
+'use strict'
+
+const path = require('path')
+const parse = require('module-details-from-path')
+const requirePackageJson = require('../require-package-json')
+const { sendData } = require('./send-data')
+const dc = require('diagnostics_channel')
+
+const savedDependencies = []
+const detectedDependencyNames = new Set()
+const FILE_PATH_START = `file:${path.sep}${path.sep}`
+const moduleLoadStartChannel = dc.channel('dd-trace:moduleLoadStart')
+
+let timeout, config, application, host
+
+function waitAndSend (config, application, host) {
+  if (!timeout) {
+    timeout = setTimeout(() => {
+      timeout = null
+      if (savedDependencies.length > 0) {
+        const dependencies = savedDependencies.splice(0, 1000)
+        sendData(config, application, host, 'app-dependencies-loaded', { dependencies })
+        if (savedDependencies.length > 0) {
+          waitAndSend(config, application, host)
+        }
+      }
+    }, 1000)
+    timeout.unref()
+  }
+}
+
+function onModuleLoad (data) {
+  if (data) {
+    const { filename, request } = data
+    if (filename && request && isDependency(filename, request) && !detectedDependencyNames.has(request)) {
+      detectedDependencyNames.add(data.request)
+      const parseResult = parse(filename)
+      if (parseResult) {
+        const { name, basedir } = parseResult
+        if (basedir) {
+          try {
+            const { version } = requirePackageJson(cleanPath(basedir), module)
```

Comment:
The versions are already captured elsewhere, so the logic should be centralized.

Reply:
ok, updated, I have created new `requirePackageVersion` method and call to it in all places that I've found that we were calling to `requirePackageJson` to get only the version.

Reply:
What I meant is that we're running the code twice to get the version. Ideally we'd either store that information somewhere, or include it in a diagnostics channel event.

Reply:
Actually there is already `loadChannel` that provides most of this information, could it just be augmented with the specific and then it would have everything you need?

Reply:
> What I meant is that we're running the code twice to get the version. Ideally we'd either store that information somewhere, or include it in a diagnostics channel event.

Ok, understood. We are parsing the file twice, one in `register.js` and other here. I will undo that small refactor... I wasn't very sure about it. I'll check what is the best way to prevent it, cache or channel.

> Actually there is already` loadChannel` that provides most of this information, could it just be augmented with the specific and then it would have everything you need?

`loadChannel` is only used when the loaded package is instrumented. We need all packages, not only the instrumenteds:
https://github.com/DataDog/dd-trace-js/blob/b819d037961e2408b748cbdd06af195c874973a3/packages/datadog-instrumentations/src/helpers/register.js#L33

Reply:
> loadChannel is only used when the loaded package is instrumented. We need all packages, not only the instrumenteds:

That could be refactored, and probably should be actually because there are other use cases that will need this as well. I guess if it's too difficult to do it in this PR it can be done later.

Reply:
I think that to do it correctly, we should change several things:

- When some module is loaded, any import or require (changes in `iitm.js` and `ritm.js`) _(should we ignore loads of any file in `dd-trace/`?)_
    - get the version of the module from the closest `package.json` and cache it
    - launch event in `dd-trace:instrumentation:load` with `{ name, version, file, moduleExports }`
- Change the current `Hook` method in `packages/datadog-instrumentations/src/helpers/hook.js` to work with `dd-trace:instrumentation:load` channel

That logic will simplify `onModuleLoad` method of `dependencies.js`, because we will have version already loaded.

I think it would be worth create a new PR